common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -151,15 +152,27 @@ private void failRemainingBlocks(String[] failedBlockIds, Throwable e) {
     }
   }
 
+  private static synchronized boolean renameFile(File src, File dest) {
+    if (dest.exists()) {
+      if (!dest.delete()) {
```

Comment:
I was using `synchronized static` because there could be multiple `OneForOneBlockFetcher` doing the rename work.